bson/value_reader.go

```diff
@@ -57,14 +72,26 @@ func newValueReader(t Type, r io.Reader) ValueReader {
 }
 
 func newDocumentReader(r io.Reader) *valueReader {
```

Comment:
`newDocumentReader()` is also called where `releaseDocumentReader()` is not called. It seems we will have to reverse `newDocumentReader()` and add a `reset()` for `valueReader` like what we did for `valueWriter`.

Reply:
Good catch!